### DISCUSSION TITLE
How to connect to a Minio server that is not hosted on same kubernetes cluster

### DISCUSSION THREAD

**Discussion by jzsiggy** [argoproj/argo-workflows#5703] [Q&A]
Is there a way to connect my workflow artifact repository to a Minio server that is being hosted on a separate machine and not in the same kubernetes cluster as the one from where I am running my workflow? 

All the tutorials that I've seen demonstrate how to set up an artifact repository on Minio when it's being served on the same kubernetes cluster. I would like, however, to set up my artifact repository on a Minio server that is running on a separate machine that is not in the kubernetes cluster (but is in my local network). After I set up my configmap and add the minio credentials I get the following error when I try to save an artifact:

`msg="Failed to put file: Get \"http://172.17.0.2:9000/test/?location=\": dial tcp 172.17.0.2:9000: connect: connection refused"`

Has anyone ran into this error before or know how to get this to work? Thanks!

**Comment by terrytangyuan** [comment]
Your Minio server is probably configured only to be accessible within your cluster. If you have applications in other clusters that want to access it, you need to configure the appropriate network rules to expose access to the Minio service. Kubernetes provides multiple options for configuring external access to services. Some reference documentations on how to configure this:
* https://kubernetes.io/docs/concepts/services-networking/service/#publishing-services-service-types
* https://kubernetes.io/docs/concepts/services-networking/ingress/

**Reply by jzsiggy** [reply]
Hey Terry, Thanks! Actually, my minio server isn't installed in a kubernetes cluster at all. It is running as a docker image that I started with the command:
`docker run -p 9000:9000 minio/minio server /data`

From my browser I can access the minio gui just fine, It just won't connect when I try to access it from the argo workflow... Any Ideas?
